Update First-Installation.md

GregariousJB opened the pull request:
## Installation

* If the number of raspberries shown at boot is 4, it could mean a Pi 2, 3, or 4, of which there are two different downloads for, which makes the graphic somewhat useless/confusing. I'm replacing this with some more concrete methods of discovering Pi model.
* While writing the instructions for the SD card reader, I realized it's really better off in the "Required" section, because the user really needs *some* way of plugging it into their PC for a first install, so I fixed up the Hardware section a bit.

I was going to be gentle with this Installation section and maybe have a new "Quick Setup" headered section right after Hardware, but then I thought... fuck it. I'll just rewrite the whole thing. 

I'm leaning heavily on the Raspberry Pi Imager as discussed in previous PRs. If it does all the checksums and it's an official RPI program and it's mostly future-proof, then this is literally all that's needed. We don't need to download the image separately, then extract it, etc etc. Just use the damn program. Done.

Because this program is the main focus of first-time setups, its instruction needs to be solid. I opted to write out the instructions *and* include a short gif showing how the program is used for those of us that really preferred pop-up books growing up and have no patience for long videos or walls of text.

The advantage is, aside from the above "Video Cable" list which may or may not change in the near future (maybe DisplayPort for the RPI 5?), this entire page is now completely future-proof and barely needs to be touched after this merge. Most everything that can change is linked outside of this page.

Let me know if there's anything I can improve.

dankcushions commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,57 @@ The simplest way to get most of these components is through an all-in-one kit su
-   * The [Control Block](http://blog.petrockblock.com/2014/12/29/controlblock-power-switch-and-io-for-the-raspberry-pi/) can use original SNES controllers
+* [Raspberry Pi](https://retropie.org.uk/about/building/)
+* MicroSD Card ([list of compatible SD cards](http://elinux.org/RPi_SD_cards))
+* [USB MicroSD card reader](https://user-images.githubusercontent.com/540857/107463579-68728200-6b2c-11eb-9612-05c4d1931882.jpg) - an adapter that allows plugging the MiscroSD card into any USB port
```
typo "MiscroSD" but IMO title should be "MicroSD card reader" and instead
> * MicroSD card reader ([Example #1](https://user-images.githubusercontent.com/540857/107868575-6f163780-6e53-11eb-9943-faec57cbf48e.png), [example #2](https://user-images.githubusercontent.com/540857/107868592-94a34100-6e53-11eb-91ed-e645be9827aa.jpg)) - If your computer/laptop does not have an MicroSD card slot, acquire an [external USB MicroSD card reader](https://user-images.githubusercontent.com/540857/107463579-68728200-6b2c-11eb-9612-05c4d1931882.jpg).

It just reads better to me to have the no-effort part before the effort part :)

dankcushions commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,57 @@ The simplest way to get most of these components is through an all-in-one kit su
-If you get the error `Illegal Instruction` when it boots or if it just boots into the terminal, you picked the wrong SD image or the image was corrupted on download or extraction.
+The [MicroSD card](https://user-images.githubusercontent.com/540857/107868231-830c6a00-6e50-11eb-85d1-9a6ada5f28b4.jpg) needs to be plugged into your PC, but most PCs don't have a port for it (although some laptops do). You'll need to insert the MicroSD card into an adapter, such as a [USB MicroSD card reader](https://user-images.githubusercontent.com/540857/107463579-68728200-6b2c-11eb-9612-05c4 ...
+### Download
+Download the official [Raspberry Pi Imager](https://www.raspberrypi.org/software/), which allows downloading the different OS versions of Retropie.
```
I'll defer to @mitu et al here but I guess I don't love the reliance on a specific third party tool here, even if it is the Raspberry Pi one. also, only supports osx, ubuntu and windows (although that probably covers 99% of users, and those using other OSs will surely be able to figure it out). but i will go with consensus! it certainly simplifies the guide. i think i have convinced myself it's a good move in writing this comment :)

GregariousJB replied:
Yeah, @cmitu wants the old instruction added back in as a secondary option, which is probably a good idea. I'll add it to an "Alternative Installation Methods" section or something.

dankcushions commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,57 @@ The simplest way to get most of these components is through an all-in-one kit su
-|![rpi1](https://cloud.githubusercontent.com/assets/10035308/21957849/f64fe008-da54-11e6-9a1c-09cb9e87c8f5.png) | ![rpi2](https://cloud.githubusercontent.com/assets/10035308/21957850/f66f2120-da54-11e6-876b-6d7f276c31e3.png)|
-
-If you get the error `Illegal Instruction` when it boots or if it just boots into the terminal, you picked the wrong SD image or the image was corrupted on download or extraction.
+The [MicroSD card](https://user-images.githubusercontent.com/540857/107868231-830c6a00-6e50-11eb-85d1-9a6ada5f28b4.jpg) needs to be plugged into your PC, but most PCs don't have a port for it (although some laptops do). You'll need to insert the MicroSD card into an adapter, such as a [USB MicroSD card reader](https://user-images.githubusercontent.com/540857/107463579-68728200-6b2c-11eb-9612-05c4 ...
```
We've already covered using/getting a reader above so I think this should be reduced to:

> Insert the [MicroSD card](https://user-images.githubusercontent.com/540857/107868231-830c6a00-6e50-11eb-85d1-9a6ada5f28b4.jpg) into the reader.

dankcushions commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,57 @@ The simplest way to get most of these components is through an all-in-one kit su
-**Note** RetroPie is built on top of Raspberry Pi OS Buster (a Linux based OS for the Raspberry Pi) and as such the partition on the SD card is EXT4 (a linux filesystem). This partition is is not visible on Windows systems, so the card will show up as a smaller size than usual and you won't be able to see everything on the card, but it is all there. You will be able to access the filesystem over  ...
 
-If you're updating from a previous version of RetroPie see [**HERE**](Updating-RetroPie).
+**Note**: You can safely ignore Windows' warnings about this drive. You may see Windows complain that "There is a problem with this drive", but everything is fine. RetroPie is built on top of Raspberry Pi OS Buster (a Linux based OS for the Raspberry Pi) and as such the partition on the SD card is EXT4 (a linux filesystem). This partition is not visible on Windows systems, so the card will show u ...
```
this part is windows-specific. I am not sure it's necessary in any case.

cmitu replied:
The warning doesn't show up with an empty microSD, on first installation. It happens after writing the image or if the microSD has an existing RetroPie/Raspbian. I don't think it's needed.

GregariousJB replied:
First time I installed Retropie I thought I messed something up and plugged the MicroSD back into my computer, then saw that message on Windows 10. As a result, I followed the message through to what I believe was a GUI version of `chkdsk`. Windows reported the drive was "fixed" (although it probably did nothing), but I was still getting the error and so ignored it. 

I figure having this line could help reduce forum posts about it, but if you aren't getting any to begin with I'll just remove it.

dankcushions commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,57 @@ The simplest way to get most of these components is through an all-in-one kit su
+* [Raspberry Pi](https://retropie.org.uk/about/building/)
+* MicroSD Card ([list of compatible SD cards](http://elinux.org/RPi_SD_cards))
+* [USB MicroSD card reader](https://user-images.githubusercontent.com/540857/107463579-68728200-6b2c-11eb-9612-05c4d1931882.jpg) - an adapter that allows plugging the MiscroSD card into any USB port
+  * Some laptops have this functionality in the form of an SD card slot ([example #1](https://user-images.githubusercontent.com/540857/107868575-6f163780-6e53-11eb-9943-faec57cbf48e.png), [example #2](https://user-images.githubusercontent.com/540857/107868592-94a34100-6e53-11eb-91ed-e645be9827aa.jpg)). In this case, a USB MicroSD card reader may not be needed, but can still be useful to have to a ...
```
> In this case, a USB MicroSD card reader may not be needed, but can still be useful to have to allow manipulation of the MicroSD card in any computer with a USB port.

I don't get this part. If you have a reader, you don't need to buy one :)

GregariousJB replied:
My thought process was that a USB card reader can be used in a laptop or desktop, while the built-in card reader on a laptop is only useful for that laptop.  Therefore, having the USB option makes it more versatile. Maybe there's some potential confusion between a MIcroSD card reader and a USB MicroSD card reader if people assume they're the same thing.

cmitu commented:
> * If the number of raspberries shown at boot is 4, it could mean a Pi 2, 3, or 4, of which there are two different downloads for, which makes the graphic somewhat useless/confusing. I'm replacing this with some more concrete methods of discovering Pi model.

Yes, this part of identifying has become kind of obsolete, recent kernel changes don't show the berries on the Pi4. Plus there's now the Pi 400, which is a computer on its own and can easily be identified.
The link to Wikipedia is not so user friendly, but it's better than what we had, I guess it's ok.
 
> * While writing the instructions for the SD card reader, I realized it's really better off in the "Required" section, because the user really needs _some_ way of plugging it into their PC for a first install, so I fixed up the Hardware section a bit.
> 

Looks fine, but withe notes added.

> I'm leaning heavily on the Raspberry Pi Imager as discussed in previous PRs. If it does all the checksums and it's an official RPI program and it's mostly future-proof, then this is literally all that's needed. We don't need to download the image separately, then extract it, etc etc. Just use the damn program. Done.

Please leave the other options listed, with RPI Imager as preferred option. They're still valid programs and Win32DiskImager is the only one that can do backups (which is a common question in the forums).
We could mention it's possible to skip the download step and download RetroPie directly from the RPI imager (as a bonus). As said in previous PR, link to the RPI video for detailed instructions if you wish, but writing the image does not need a video/gif here in the page.

GregariousJB commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,58 @@ The simplest way to get most of these components is through an all-in-one kit su
+* [MicroSD card](https://user-images.githubusercontent.com/540857/107868231-830c6a00-6e50-11eb-85d1-9a6ada5f28b4.jpg) ([list of compatible SD cards](http://elinux.org/RPi_SD_cards))
+* MicroSD card reader - a way to plug the MicroSD card into your computer or laptop
+  * Some laptops have this functionality in the form of an SD card slot ([example #1](https://user-images.githubusercontent.com/540857/107868575-6f163780-6e53-11eb-9943-faec57cbf48e.png), [example #2](https://user-images.githubusercontent.com/540857/107868592-94a34100-6e53-11eb-91ed-e645be9827aa.jpg)).
+  * A [USB MicroSD card reader](https://user-images.githubusercontent.com/540857/107463579-68728200-6b2c-11eb-9612-05c4d1931882.jpg) can be plugged into any USB port
```
@dankcushions I ended up slightly re-writing this section for clarity. Let me know if you agree with the change.

dankcushions replied:
I think 'Computer (or Laptop)' should be part of the 'Required' items, and then all references to them after should simply be 'Computer' ('laptop' is a subtype of Computer, and tower-type computers can have SD card slots anyway)

GregariousJB commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,58 @@ The simplest way to get most of these components is through an all-in-one kit su
+6. Select the image for the model of Raspberry Pi (RPI) that you have. For example, if you have a Raspberry Pi 4, select the RPI 4/400 option ([example](https://user-images.githubusercontent.com/540857/107868009-493a6400-6e4e-11eb-9272-7f45d569dc44.gif)) 
+   * If you don't know which model Raspberry Pi you have, there are a few methods to find out: 
+     * The [Raspberry Pi Wikipedia page](https://en.wikipedia.org/wiki/Raspberry_Pi#Connectors) has user-friendly graphics to help determine model by looking at the board itself 
+     * If you have SSH/commandline access, run `cat /proc/device-tree/model` and it will output your Pi's model
```
**These nested list sections are probably going to break!**

I noticed it happens only if you switch from numbers to bullet points - MD has to have 3 spaces after the switch or it gets ignored, but that Material extension doesn't like 3 spaces. I'm leaving this formatted for MD for my sanity so I can do these edits and see the end result.

cmitu replied:
We can fix those after. But you can't mix both list types due to the `mkdocs` extension for lists.

GregariousJB replied:
@cmitu Do you mean the _mdx_truly_sane_lists_ extension? 

So it limits spaces from 4 to 2 for nested lists _and_ it messes with mixing ordered/unordered nested lists? Does it have an upside?

cmitu replied:
Yes, it's easier to write without producing excessive white space between list items.

GregariousJB commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,58 @@ The simplest way to get most of these components is through an all-in-one kit su
+2. Download a program to write the image to your SD card.
+   * For Windows: [Raspberry Pi Imager](https://www.raspberrypi.org/software/), [Etcher](https://www.balena.io/etcher/), or [Win32DiskImager](https://sourceforge.net/projects/win32diskimager/)
+   * For macOS: [Raspberry Pi Imager](https://www.raspberrypi.org/software/), [Etcher](https://www.balena.io/etcher/), or [Apple Pi Baker](https://www.tweaking4all.com/hardware/raspberry-pi/applepi-baker-v2/)
+   * For Linux: [Raspberry Pi Imager](https://www.raspberrypi.org/software/), [Etcher](https://www.balena.io/etcher/), or the `dd` command
```
**More list breakage here**

GregariousJB commented:
Ok, I've made most of both of your corrections and tweaked one or two others. 

And, _fine_, I moved the gif from an embedded image to a link. 😢 Although, it does make the instruction quite a bit shorter.

I removed the "Plug in a MicroSD" section because it felt silly having a whole headered section with a one-line instruction.

I have a concern about how I've done the Alternative Method, which is now the "old" installation instructions that I've added back in - With so many programs listed there's just no way to specify what the user needs to do without it taking 3 pages of instructions. And I just threw in the part about using 7zip to extract the .img file because that sounded important to you, @cmitu. I'm at a loss for how you want this handled.

dankcushions commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,58 @@ The simplest way to get most of these components is through an all-in-one kit su
+1. [Download the RetroPie image](https://retropie.org.uk/download/) from the official website for your Raspberry Pi model.
+2. Download a program to write the image to your SD card.
+   * For Windows: [Raspberry Pi Imager](https://www.raspberrypi.org/software/), [Etcher](https://www.balena.io/etcher/), or [Win32DiskImager](https://sourceforge.net/projects/win32diskimager/)
+   * For macOS: [Raspberry Pi Imager](https://www.raspberrypi.org/software/), [Etcher](https://www.balena.io/etcher/), or [Apple Pi Baker](https://www.tweaking4all.com/hardware/raspberry-pi/applepi-baker-v2/)
```
macOs can also use `dd`

GregariousJB replied:
For some reason I can't comment on your above remark about listing a computer as a required item - Agreed.

dankcushions commented on docs/First-Installation.md:
```diff
@@ -21,65 +21,58 @@ The simplest way to get most of these components is through an all-in-one kit su
+3. Click the **Choose OS** button
+4. Select **Emulation and game OS**
+5. Select **RetroPie**.
+6. Select the image for the model of Raspberry Pi (RPI) that you have. For example, if you have a Raspberry Pi 4, select the RPI 4/400 option ([example](https://user-images.githubusercontent.com/540857/107868009-493a6400-6e4e-11eb-9272-7f45d569dc44.gif)) 
```
I believe @cmitu 's comment was suggesting still to do this via downloading and selecting an image (with the 'shortcut' as a breakout note), so these two methods could be combined into one, with Raspberry Pi Imager as the 'preferred' program to write the image. I feel this way would make this section tidier, rather than having 2 'methods'

cmitu replied:
Yes, I was thinking something along the lines _if you use Raspberry Pi Imager, you can omit the image download and choose RetroPie from the list of available images_.

GregariousJB replied:
Oh, yeah I might have misunderstood. @cmitu?

GregariousJB commented:
Ok, I combined the two methods and just emphasized using RPI Imager. I do like this a lot more. I don't feel like I need to repeat instructions between the two methods, and it's a helluva lot shorter.

It's still a little generic in step #3 due to all the different programs listed, but I included the [official RPI Imager instruction](https://www.raspberrypi.org/documentation/installation/installing-images/) link present in the current First Installation page, so hopefully that will suffice.

Thoughts?

cmitu commented:
Looks fine. Just 2 things
- in macOS you can double click on the `gz` and it's automatically extracted, so you can skip the `gunzip` part for it. 
- don't mention Buster as the release for Raspi OS, this will change.


Formatting wise, there's a bit of extra nesting in list item 2, for the image selection. Just merge the 2 paragraphs and leave a single sub-list with the 'find out methods'
```
   * Select the image for the model of Raspberry Pi (RPI) that you have. For example, if you have a Raspberry Pi 4, select the *RPI 4/400* image.
    If you don't know which model Raspberry Pi you have, there are a few methods to find out: 
     * The [Raspberry Pi Wikipedia page](https://en.wikipedia.org/wiki/Raspberry_Pi#Connectors) has user-friendly graphics to help determine model by looking at the board itself
     * If you have SSH/commandline access, run `cat /proc/device-tree/model` and it will output your Pi's model
```

GregariousJB commented:
Ok, cool. I think I understood. How does that look?

cmitu commented:
Not exactly what I had in mind, but it's ok.
Thank you.

GregariousJB commented:
If it's something I can fix, even small, by all means let me know. This is the kind of page worth being picky and thorough.

Also, just a friendly reminder to fix those nested lists. Sorry to have to dump that on you. I'd prefer to be able to have it all done when you decide to merge it.
